application/dash_application/utility/plot_builder.py

```diff
@@ -769,4 +772,247 @@ def items_for_df(self):
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            name_col,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y
+    ):
+        self.title = title
+        self.y_label = y_label
+        self.df = df
+        self.x_col = x_col
+        self.y_col = y_col
+        self.colourby = colourby
+        self.shapeby = shapeby
+        self.hovertext_cols = hovertext_cols
+        self.cutoff_lines = [] if cutoff_lines is None else cutoff_lines
+        self.name_col = name_col
+        self.markermode = markermode
+        self.bar_positive = bar_positive
+        self.bar_negative = bar_negative
+        self.log_y = log_y
+
+    def traces(self):
+        return _generate_traces(
+            self.df,
+            self.x_col,
+            self.y_col,
+            self.colourby,
+            self.shapeby,
+            self.hovertext_cols,
+            self.cutoff_lines,
+            self.name_col,
+            self.markermode,
+            self.bar_positive,
+            self.bar_negative,
+        )
+
+    def is_pct_graph(self):
+        return self.y_label == GraphTitles.PCT
+
+
+class SingleLaneSubplot(Subplot):
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines=None,
+            name_col=PINERY_COL.SampleName,
+            markermode="markers",
+            bar_positive=None,
+            bar_negative=None,
+            log_y=False
+    ):
+        super().__init__(
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            name_col,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y,
+        )
+
+
+class CallReadySubplot(Subplot):
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines=None,
+            name_col=PINERY_COL.SampleName,
+            markermode="markers",
+            bar_positive=None,
+            bar_negative=None,
+            log_y=False,
+    ):
+        super().__init__(
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            name_col,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y,
+        )
+
+
+class GraphTitles:
```

Comment:
Enum would make more sense. These are currently not used. I think Heather's idea was to have a central repository of graph titles. i am not sure if that needs to be done or if this is the best way.

Reply:
Seems to me like it's not really solving any problem - we once wanted to rename all the Total Reads graphs but we've since done that and it's all been fine, we haven't needed to rename en masse

Reply:
Left a TODO. I think this feeds into Dashi giving glossary information, but that is a whole bigger fish.